docs/site/tutorials/todo-list/todo-list-tutorial-docker-mysql.md

```diff
@@ -0,0 +1,237 @@
+---
+lang: en
+title: 'Setting up MySQL in Docker'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, TypeScript, OpenAPI, Tutorial
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/todo-list-tutorial-docker-mysql.html
+summary: LoopBack 4 TodoList Application Tutorial - Setting up MySQL in Docker
+---
+
+Here are the instructions for getting MySQL running in a docker container for
+the purposes of running the TodoList example against a relational database.
+
+## Install Docker
+
+For downloading and installing docker, see
+[Get Docker](https://docs.docker.com/get-docker/) for instructions on how to
+install it on your respective operating system.
+
+## Get MySQL docker image
+
+```sh
+docker pull mysql/mysql-server:latest
+```
+
+## List MySQL docker image
+
+```sh
+docker images
+```
+
+outputs:
+
+```sh
+REPOSITORY TAG IMAGE ID CREATED SIZE
+mysql/mysql-server latest 8a3a24ad33be 2 months ago 366MB
+```
+
+## Create a docker container based on the image, and set root password and port
+
+We will give the docker container a name of `todo-mysql` and we will give the
+`root` user a password of `my-secret-pw`.
+
+```sh
+docker run --name todo-mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=my-secret-pw  -d mysql/mysql-server
+```
+
+outputs a UUID for the container:
+
+```sh
+191508fcbcfd3b683fea06bb3c8a2f1f252527a15891b58b124518cf2cb60249
+```
+
+If the operating system complains that your 3306 port is already being used on
+your machine, you can remove the container, and try a different port number.
+
+```sh
+docker rm todo-mysql
+docker run --name todo-mysql -p 3307:3306 -e MYSQL_ROOT_PASSWORD=my-secret-pw -d mysql/mysql-server
+```
+
+## View the status of your docker container on your machine
+
+```sh
+docker ps
+
+OR
+
+docker ps -a
+```
+
+The first lists running containers, and the second lists ALL containers (running
+and stopped).
+
+If it's up and running, the output should look like this:
+
+```sh
+CONTAINER ID        IMAGE                COMMAND                  CREATED             STATUS                   PORTS                               NAMES
+24505c80a83d        mysql/mysql-server   "/entrypoint.sh mysq…"   7 minutes ago       Up 7 minutes (healthy)   0.0.0.0:3306->3306/tcp, 33060/tcp   todo-mysql
+```
+
+## Fixing the root user info in the database
```

Comment:
Can anything be done from our end to avoid having to perform these commands to switch MySQL to native authentication? Can our connector settings be adjusted to connect properly with this sha-2 approach?

Reply:
> Can anything be done from our end to avoid having to perform these commands to switch MySQL to native authentication? Can our connector settings be adjusted to connect properly with this sha-2 approach?

I noticed that our loopback-connector-mysql depends on the node.js `mysql` driver, which currently doesn't support the new authentication method in MySQL 8.0. See : https://github.com/strongloop/loopback-connector-mysql/blob/master/setup.sh#L51 and https://github.com/mysqljs/mysql/issues/2002 . This is why Node.js applications need to go through the steps to set the default authentication method in MySQL 8.0 back to `native`. 
I will revise our tutorial to use MySQL 5.7.22, but will provide a section on what they can do if they insist on using 8.0